EIPS/eip-6672.md

```diff
@@ -0,0 +1,247 @@
+        },
+        "image": {
+            "type": "string",
+            "description": "A URI pointing to a resource with mime type image/* representing the asset to which this NFT represents. Consider making any images at a width between 320 and 1080 pixels and aspect ratio between 1.91:1 and 4:5 inclusive."
+        }
+    },
+    "redemptions": {
+        "operator-tokenId-redemptionId": {
+            "status": {
+                "type": "string",
+                "description": "The status of a redemption. Enum type can be used to represent the redemption status, such as redeemed, shipping, paid."
+            },
+            "description": {
+                "type": "string",
+                "description": "Describes the object that has been redeemed for an NFT, such as the name of an action figure series name or the color of the product."
+            }
+        }
+    }
+}
+```
+
+## Rationale
+
+### Redeem and Cancel Mechanism
+
+An operator SHALL only make an update to the redemption created by itself. Therefore, the `redeem()` and `cancel()` functions do not have an `_operator` parameter, and the `msg.sender` address MUST be used as the `_operator`.
+
+### Key Choices for Redemption Flag
+
+The combination of `_operator`, `_tokenId`, and `_redemptionId` is REQUIRED to be used as the key in the redemption flag key-value pairs. The reasoning is as follows:
+
+- Operator wallet address, i.e. `_operator`
+
+    It's possible that there are more than one party who would like to use the same NFT for redemption. For example, MisterPunks NFTs are eligible to be redemeed for both Event-X and Event-Y tickets, and each event's ticket redemption is handled by a different operator.
+
+- Token identifier, i.e. `_tokenId`
+
+    Each NFT holder will have different redemption records created by the same operator. Therefore, it's important to use token identifier as one of the keys.
+
+- Redemption identifier, i.e. `_redemptionId`
+
+    Using `_redemptionId` as one of the keys enables NFT holders to redeem the same NFT to the same operator in multiple campaigns. For example, Operator-X has 2 campaigns, i.e. campaign A and campaign B, and both campaigns allow for MisterPunks NFTs to be redemeed for physical action figures. Holder of MisterPunk #7 is eligible for redemption in both campaigns and each redemption is recorded with the same `_operator` and `_tokenId`, but with different `_redemptionId`.
+
+### Metadata Choices (metadata extension)
+
+To provide additional information about a redemption, `redemptions` key-value pairs is added to the "[ERC-721](./eip-721.md) Metadata Extension". The key format for the redemptions key-value pairs MUST be standardized as `operator-tokenId-redemptionId`, where `operator` is the operator wallet address, `tokenId` is  the identifier of the token that has been redeemed, and `redemptionId` is the redemption identifier. The same combination of keys is used for the redemption flag, and the same rationale applies.
```

Comment:
Seems like there's another requirement in here (the key-value format) that needs to be in Specification.

Reply:
moved to Specification